modules/inject-io/src/main/scala/io/janstenpickle/trace4cats/inject/io/package.scala

```diff
@@ -0,0 +1,5 @@
+package io.janstenpickle.trace4cats.inject
+
+import _root_.io.janstenpickle.trace4cats.base.context.io.IOLocalContextInstances
+
+package object io extends IOLocalTraceInstances with IOLocalContextInstances
```

Comment:
The usage will be as follows:
```scala
package my.app

import cats.effect.{IO, IOApp, IOLocal}
import io.janstenpickle.trace4cats.Span
import io.janstenpickle.trace4cats.base.context.Provide
import io.janstenpickle.trace4cats.inject.{EntryPoint, Trace}
import io.janstenpickle.trace4cats.kernel.{SpanCompleter, SpanSampler}

// this imports both provide and trace instances for IOLocal
import io.janstenpickle.trace4cats.inject.io._ 

object Test extends IOApp.Simple {
  def run: IO[Unit] = {
    val spanCompleter = SpanCompleter.empty[IO]
    Span.discharge(SpanSampler.always[IO], spanCompleter).flatMap { span =>
      IOLocal(span).map { ioLocal =>
        implicit val trace: Trace[IO] = ioLocalTrace(ioLocal)
        implicit val provide: Provide[IO, IO, Span[IO]] = ioLocalProvide(ioLocal)
        Fibonacci(EntryPoint[IO](SpanSampler.always[IO], spanCompleter))
      }
    }
  }
}
```

Reply:
Looks good, I'll try it on the demo repo later!

Once this PR is merged I'd like to experiment with adding some helper for this as good parts of this snippet will always be the same, so we can make user code nicer by introducing some `IOLocalHelper` (better name to be found) that creates the discharge span and the `IOLocal` and returns type class instances ready to use. Especially on Scala 3 (context functions), this could simplify usage noticeably.

Reply:
@ybasket We can include a syntax extension for `Span.type`. So it may look as:
```scala
val x = Span.dischargeIOLocal(sampler, completer) : IO[IOLocal[Span[IO]]
```

Reply:
If you don't mind, let's put it in a separate PR? I just want to release something we can play with.

Reply:
Yes, definitely follow-up PR(s) for anything that's not in here yet.